ert3/engine/_run.py

```diff
@@ -34,35 +32,34 @@ def _add_record(records, record_name, record):
         ens_records[record_name] = new_record
 
 
-def _load_input_records(workspace_root, experiment_name):
-    with open(workspace_root / experiment_name / "ensemble.yml") as f:
-        ensemble = yaml.safe_load(f)
-
-    ensemble_size = ensemble["size"]
+def _load_input_records(ensemble, workspace_root, experiment_name):
 
-    records = [{} for _ in range(ensemble_size)]
-    for input_record in ensemble["input"]:
-        record_name = input_record["record"]
-        record_source = input_record["source"].split(".")
+    records = [{} for _ in range(ensemble.size)]
+    for input_record in ensemble.input:
+        record_name = input_record.record
+        record_source = input_record.source.split(".")
 
         var_name = _persist_variables(
-            record_name, record_source, ensemble_size, experiment_name, workspace_root
+            record_name, record_source, ensemble.size, experiment_name, workspace_root
         )
 
         input_record = ert3.storage.get_variables(workspace_root, var_name)
         _add_record(records, record_name, input_record)
     return records
 
 
-def run(workspace_root, experiment_name):
-    ert3.workspace.assert_experiment_exists(workspace_root, experiment_name)
+def _create_forward_model(stages_config, ensemble):
```

Comment:
Can me make the assumption that the forward model contains of a single stage and script by assertions? :)

Reply:
I guess we can, I was a bit bothered by this, because it is really not a good solution as-is. I added asserts and created an issue, here: https://github.com/equinor/ert/issues/1243